fairlearn/metrics/_multi_curve_plot.py

```diff
@@ -0,0 +1,126 @@
+from ..utils._input_validation import _validate_and_reformat_labels, _validate_and_reformat_labels_and_sf, check_consistent_length
+from ..postprocessing._plotting import _MATPLOTLIB_IMPORT_ERROR_MESSAGE
+from ._make_derived_metric import _DerivedMetric
+from typing import Callable, List, Union
+
+def plot_model_comparison(
+    *,
+    x_axis_metric: Callable[..., Union[float, int]],
```

Comment:
>     * Would it be a nice convention to make a `metric` type? I guess because there is no type inference and we want to be able to use sklearn functions without having to initialize a `metric` type for them?

The sklearn metrics are all callables. I think it's best to stay compatible with sklearn (as we've done with everything so far). We could always do some checking on whether the functions have the required args, but that strikes me as somewhat unpythonic. Your arg description says quite clearly what's needed, and if you don't know the function you should really read the docs, right? Perhaps I'm too optimistic here 🤣

Reply:
Ahh yes that makes sense!